packages/flutter/lib/src/widgets/icon_theme.dart

```diff
@@ -21,7 +21,7 @@ import 'inherited_theme.dart';
 /// Controls the default properties of icons in a widget subtree.
 ///
 /// The icon theme is honored by [Icon] and [ImageIcon] widgets.
-class IconTheme extends InheritedTheme {
+class IconTheme extends InheritedTheme<IconThemeData, Object?> {
```

Comment:
Now I guess developers should typically avoid IconTheme.of unless they're using the whole IconTheme, and otherwise use InheritedModel.inheritFrom<InheritedTheme<IconThemeData,  Object?>>`? Can you add something to the docs of IconTheme.of to point people in that direction?

Same for each other InheritedTheme subclass.

Reply:
Sure, I will do.

Reply:
Would it be a bad idea to add `select` method to all InheritedTheme subclass ?